### PR TITLE
[UI] Change UI to send json request for user preferences

### PR DESCRIPTION
**Pull request by Abhishek-kumar09**
**Description**
Change port for provider Auth

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Abhishek-kumar09** [comment]
user_preferences tests is all passed:
image

**Comment by Abhishek-kumar09** [comment]
I think this draft doesn't need review rn, considering there are no progress in the PR other than the same Rudraksh's commit. I will mark it open once I am done.

**Comment by Abhishek-kumar09** [comment]
Changed the UI to send pref request as json. Getting 200 Now :+1:

**Comment by leecalcote** [comment]
@Abhishek-kumar09 is it expected that the e2e test is failing in this PR? I assume that it will continue to fail unto Rudraksh's other PR is merged, is that right?

**Comment by Abhishek-kumar09** [comment]
No @leecalcote 
This PR contains all the @DelusionalOptimist changes. So it is not expected to fail.
Actually user preferences UI and API is working fine and as expected if the user is logged in with meshery-provider. 

**It is None provider due to which server panics.** as discussed.

File: handlers/error.go [hunk 1]
```diff
@@ -79,6 +79,7 @@ const (
 	ErrCreateDirCode            = "2150"
 	ErrInvalidRequestObjectCode = "2151"
 	ErrChangeK8sContextCode     = "2152"
+	ErrSavingUserPreferenceCode = "some_code"
```
Comment (bariqhibat) [inline handlers/error.go:82]: by pattern, we should have this "2153"

File: handlers/user_handler.go [hunk 2]
```diff
@@ -52,109 +51,64 @@ func (h *Handler) UserPrefsHandler(w http.ResponseWriter, req *http.Request, pre
...
+		h.log.Error(ErrSavingUserPreference(err))
+		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	gen := req.FormValue("gen")
-	genTrack := false
-	// TODO: after we have interfaces for load generators in place, we need to make a generic check, for now using a hard coded one
+
+	loadGen := prefObj.LoadTestPreferences.LoadGenerator
+	loadGenSupoorted := false
```
Comment (bariqhibat) [inline handlers/user_handler.go:93]: nitpick: some typo, should be `loadGenSupported`

**Review by bariqhibat — COMMENTED** [review]
Few comments

File: handlers/error.go (hunk 1 above)
Comment (leecalcote) [reply handlers/error.go:82]: Will this be automatically filled in by the area code utility, though?

File: handlers/user_handler.go (hunk 2 above)
Comment (leecalcote) [reply handlers/user_handler.go:93]:
```suggestion
	loadGenSupported := false
```

File: handlers/user_handler.go [hunk 3]
```diff
@@ -52,109 +51,64 @@ func (h *Handler) UserPrefsHandler(w http.ResponseWriter, req *http.Request, pre
...
-	genTrack := false
-	// TODO: after we have interfaces for load generators in place, we need to make a generic check, for now using a hard coded one
+
+	loadGen := prefObj.LoadTestPreferences.LoadGenerator
+	loadGenSupported := false
 	for _, lg := range []models.LoadGenerator{models.FortioLG, models.Wrk2LG, models.NighthawkLG} {
-		if lg.Name() == gen {
-			genTrack = true
+		if lg.Name() == loadGen {
+			loadGenSupoorted = true
```
Comment (manav1403) [inline handlers/user_handler.go:96]: variable name typo
```suggestion
			loadGenSupported = true
```

File: handlers/user_handler.go [hunk 4]
```diff
@@ -52,109 +51,64 @@ func (h *Handler) UserPrefsHandler(w http.ResponseWriter, req *http.Request, pre
...
 		}
 	}
-	if !genTrack {
-		logrus.Error("invalid value for gen")
-		http.Error(w, "please provide a valid value for gen (load generator)", http.StatusBadRequest)
+	if !loadGenSupoorted {
```
Comment (manav1403) [inline handlers/user_handler.go:99]:
```suggestion
	if !loadGenSupported {
```

**Review by manav1403 — CHANGES_REQUESTED** [review]
@Abhishek-kumar09 can u verify that after these changes is the issue resolved

**Pull request merged**